File: benchmarks/lua/fib-recursive.lua
-- benchmark: fib-recursive (LuaJIT)
-- Naive recursive Fibonacci, fib(31) = 1346269. The seed is read from a table
-- so the recursion is not constant-folded.

local function fib(n)
  if n < 2 then
    return n
  end
  return fib(n - 1) + fib(n - 2)
end

local function workload()
  local seed = { [0] = 31 }
  return fib(seed[0])
end

local WARMUP, TIMED = 3, 11
local checksum = 0
for _ = 1, WARMUP do checksum = workload() end
local times = {}
for i = 1, TIMED do
  local t0 = os.clock()
  checksum = workload()
  local t1 = os.clock()
  times[i] = t1 - t0
end
table.sort(times)
local median = times[math.floor((TIMED + 1) / 2)]
io.write(string.format("%d %.9f\n", checksum, median))

File: benchmarks/js/fib-recursive.js
// benchmark: fib-recursive (JS: runs under both jsc and node)
// Naive recursive Fibonacci, fib(31) = 1346269. The seed is read from an array
// so the recursion is not constant-folded.
"use strict";

function fib(n) {
  if (n < 2) {
    return n;
  }
  return fib(n - 1) + fib(n - 2);
}

function workload() {
  var seed = [31];
  return fib(seed[0]);
}

var emit = (typeof print === "function") ? print : console.log;
function nowMs() { return performance.now(); }
(function () {
  var WARMUP = 3, TIMED = 11, checksum = 0, i;
  for (i = 0; i < WARMUP; i++) { checksum = workload(); }
  var times = new Array(TIMED);
  for (i = 0; i < TIMED; i++) {
    var t0 = nowMs();
    checksum = workload();
    var t1 = nowMs();
    times[i] = t1 - t0;
  }
  times.sort(function (a, b) { return a - b; });
  var median = times[(TIMED - 1) >> 1];
  emit(String(checksum) + " " + (median / 1000).toFixed(9));
})();
